src/vma/dev/ring_eth_cb.cpp

```diff
@@ -121,6 +123,11 @@ void ring_eth_cb::create_resources(ring_resource_creation_info_t *p_ring_info,
 			m_single_wqe_log_num_of_strides = mp_rq_caps->min_single_wqe_log_num_of_strides;
 		}
 	}
+	// FW bug only exists in CX5 if more then 2048 (1<<11) strides in WQE
+	if (m_single_wqe_log_num_of_strides < 12 ||
+	    r_ibv_dev_attr.vendor_part_id == MLX_CX4_DEV_ID) {
+		first_call = false;
```

Comment:
Why are you not asking directly if it's an CTX5 device?

```
if (CTX5(vendor_part_id))
        mark for workaround
```

rather than

```
if (CTX4(vendor_part_id))
        mark for no work-around
```